### PR TITLE
feature(web): implement create post, view post, and post edit delete dropdown ui

### LINKED ISSUE #393: Create Post Component
# 🧱 Description
We need to display our posts

## 📋 Acceptance Criteria

- [x] There is a post component.

### PR DESCRIPTION
**Pull request by hks1444**
resolves #393 #395 #396 #404 #405

### CODE REVIEW

**Review by umitcan07 — APPROVED** [review]

File: web/public/add-fill.svg
Comment (umitcan07) [inline web/public/add-fill.svg:1]: delete this file and use icon component instead

File: web/src/components/Button.tsx [hunk 1]
```diff
@@ -20,6 +20,7 @@ export const buttonInnerRing = cva(
 					"group-active:border-cyan-600",
 				],
 				secondary: ["border-white/25"],
+				fixed: ["border-white/25"],
```
Comment (umitcan07) [inline web/src/components/Button.tsx:23]: why do we need this?
is "fixed" a variant on buttons?
If you want to extend a styling behavior of a cva component, use
for example:
`button({intent: "primary", className: "your-extra-class"})`
we do not need to have another variant here

File: web/src/components/Button.tsx [hunk 2]
```diff
@@ -113,3 +114,65 @@ export const button = cva(
 		},
 	},
 );
+
+export const button2 = cva(
```
Comment (umitcan07) [inline web/src/components/Button.tsx:118]: why do we need button2?
why not add another variant?
it does not make any sense

**Review by umitcan07 — CHANGES_REQUESTED** [review]
I'm taking my spin on the upcoming changes

File: web/src/components/Button.tsx (hunk 2 above)
Comment (hks1444) [reply web/src/components/Button.tsx:118]: Button had a relative attribute and I did not want to move relative to primary secondary destructive tetritary and repeat it four times so I created my button2. But I can change it

Comment (umitcan07) [reply web/src/components/Button.tsx:118]: no, please just add an override className, that's it. This is a perfect use case of using override class in cva.
Please delete this component, and use `button({intent: "primary", className: "absolute"})`

**Comment by umitcan07** [comment]
We have done a massive refactor, can we not include all the relevant issues here before merging this PR?

Now the component and feed looks like:

CleanShot 2024-05-16 at 13 24 28@2x

And here is the profile page at the moment. A bit rough, lets deal with those on another PR

CleanShot 2024-05-16 at 13 24 59@2x

**Review by umitcan07 — APPROVED** [review]
Let's go!
Can you see my recent changes and merge if we're good!

**Comment by umitcan07** [comment]
CleanShot 2024-05-16 at 14 01 11@2x

**Pull request merged**